### PR TITLE
Ability to switch between different Locales and Currencies in a single Store

### PR DESCRIPTION
**Pull request by damianlegawiec**
https://user-images.githubusercontent.com/55154/107916015-d1f7f380-6f65-11eb-8345-c70e8c4cee10.mov

### CODE REVIEW

**Comment by spencerwp** [comment]
This is great - I'm only seeing one problem where :supported_locales is not a valid parameter of the store_attributes.

spree/core/lib/spree/permitted_attributes.rb - 
@@store_attributes += [:supported_locales]

**Comment by flobby** [comment]
Not sure if related but on admin backend when I try to access store translations I receive this error. Might be related to globalize though.

NoMethodError - undefined method `translated_attribute_names' for #<Class:0x00007fefa2c5c580>:


I tracked it down to admin/translations_controller.rb:10 // something in the index action which is very generic.

(using master on commit 166b5434860636f7fb36beb3eaa6a2763b854077)

**Comment by damianlegawiec** [comment]
> Not sure if related but on admin backend when I try to access store translations I receive this error. Might be related to globalize though.
> 
> NoMethodError - undefined method `translated_attribute_names' for #Class:0x00007fefa2c5c580:
> 
> I tracked it down to admin/translations_controller.rb:10 // something in the index action which is very generic.
> 
> (using master on commit [166b543](https://github.com/spree/spree/commit/166b5434860636f7fb36beb3eaa6a2763b854077))

Alongside this PR I'm preparing PRs for Spree Globalize and Spree I18n ( https://github.com/spree-contrib/spree_i18n/pull/876 ) to ease the migration path, hopefully, I can finish all of it today/Monday.

**Comment by channainfo** [comment]
@damianlegawiec awesome! thank you

**Comment by damianlegawiec** [comment]
> This is great - I'm only seeing one problem where :supported_locales is not a valid parameter of the store_attributes.
> 
> spree/core/lib/spree/permitted_attributes.rb -
> @@store_attributes += [:supported_locales]

Done and done!

File: frontend/app/helpers/spree/store_helper.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+module Spree
+  module StoreHelper
+    def store_country_iso(store)
+      store ||= current_store
+      return unless store
+      return unless store.default_country
```
Comment (Vegann) [inline frontend/app/helpers/spree/store_helper.rb:6]: Maybe would be better to rewrite this 2 line like this
```suggestion
      return unless store&.default_country
```

**Review by Vegann — APPROVED** [review]

**Review by btolarz — APPROVED** [review]

**Pull request merged**